components/util/remutex.rs

```diff
@@ -0,0 +1,217 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
Well I wrote it :)

Actually, you have a point, which is that the unit tests are Apache/MIT licensed, not MPL. I'll fix that.